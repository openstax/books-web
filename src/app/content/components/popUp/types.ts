import { LinkedArchiveTree, LinkedArchiveTreeNode, LinkedArchiveTreeSection } from '../../types';

export type LocationFilters = Map<
  string,
  { section: LinkedArchiveTreeNode, children?: LinkedArchiveTreeSection[] }
>;

export type LocationFiltersWithChildren = Map<
  string,
  { section: LinkedArchiveTree, children: LinkedArchiveTreeSection[] }
>;

export interface FiltersChange<T> {
  remove: T[];
  new: T[];
}
