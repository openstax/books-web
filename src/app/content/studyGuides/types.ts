import { HighlightsSummary } from '@openstax/highlighter/dist/api';

export interface State {
  isEnabled: boolean;
  summary: HighlightsSummary | null;
}
