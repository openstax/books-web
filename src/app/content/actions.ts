import { createStandardAction } from 'typesafe-actions';
import * as highlightingActions from './highlights/actions';
import * as searchActions from './search/actions';
import { ArchivePage, Book, Params, State } from './types';

export const search = searchActions;
export const highlighting = highlightingActions;

export const openToc = createStandardAction('Content/openToc')<void>();
export const closeToc = createStandardAction('Content/closeToc')<void>();
export const resetToc = createStandardAction('Content/resetToc')<void>();

export const requestBook = createStandardAction('Content/requestBook')<Params['book']>();
export const receiveBook = createStandardAction('Content/receiveBook')<Book>();

export const requestPage = createStandardAction('Content/requestPage')<Params['page']>();
export const receivePage = createStandardAction('Content/receivePage')<
  ArchivePage & {references: State['references']}
>();

export const closeCallToActionPopup = createStandardAction('Content/closeCallToActionPopup')();
export const openCallToActionPopup = createStandardAction('Content/openCallToActionPopup')();
