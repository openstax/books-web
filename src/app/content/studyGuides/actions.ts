import { createStandardAction } from 'typesafe-actions';
import {
  CountsPerSource,
  SummaryFilters,
  SummaryHighlights,
  SummaryHighlightsPagination
} from '../highlights/types';

export const receiveSummaryStudyGuides = createStandardAction(
  'Content/StudyGuides/Summary/receive'
)<
SummaryHighlights,
  {
    pagination: SummaryHighlightsPagination,
    isStillLoading?: boolean
  }
>();

export const openStudyGuides = createStandardAction('Content/StudyGuides/Summary/open')<void>();
export const closeStudyGuides = createStandardAction('Content/StudyGuides/Summary/close')<void>();
export const loadMoreStudyGuides = createStandardAction('Content/StudyGuides/loadMore')();
export const setDefaultSummaryFilters = createStandardAction('Content/StudyGuides/Summary/setDefaultFilters')<
  Partial<SummaryFilters>
>();
export const setSummaryFilters = createStandardAction('Content/StudyGuides/Summary/setFilters')<
  Partial<SummaryFilters>
>();
export const receiveStudyGuidesTotalCounts = createStandardAction(
  'Content/StudyGuides/receiveTotalCounts'
)<CountsPerSource>();
