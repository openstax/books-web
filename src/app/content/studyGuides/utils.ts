import { HighlightColorEnum } from '@openstax/highlights-client';
import { OutputParams } from 'query-string';
import { colorFilterQueryParameterName } from '../constants';
import { colorfilterLabels } from './constants';

export const getColorFiltersFromQuery = (query: OutputParams) => {
  const queryColors = query[colorFilterQueryParameterName] as HighlightColorEnum | HighlightColorEnum[] | undefined;
  if (!queryColors) { return; }

  const colors = (Array.isArray(queryColors) ? queryColors : [queryColors])
    .filter((color) => colorfilterLabels.has(color));

  return colors.length ? colors : undefined;
};
