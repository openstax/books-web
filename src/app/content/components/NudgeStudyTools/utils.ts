import { HTMLElement } from '@openstax/types/lib.dom';
import * as Cookies from 'js-cookie';
import React from 'react';
import { useSelector } from 'react-redux';
import { useDebouncedWindowSize } from '../../../reactUtils';
import { assertDocument, remsToPx } from '../../../utils';
import { page as pageSelector } from '../../selectors';
import { hasStudyGuides, studyGuidesEnabled } from '../../studyGuides/selectors';
import {
  arrowDesktopHeight,
  arrowLeftMargin,
  arrowMobileHeight,
  arrowTopMargin,
  closeButtonDistanceFromContent,
  contentMarginTop,
  cookieNudge,
  nudgeStudyToolsMinPageLimit,
  nudgeStudyToolsShowLimit,
  nudgeStudyToolsTargetId,
  spotlightPadding,
  timeIntervalBetweenShowingNudgeInMs,
} from './constants';

interface Positions {
  arrowLeft: number;
  arrowTopOffset: number;
  closeButtonLeft: number;
  closeButtonTopOffset: number;
  contentWrapperRight: number;
  contentWrapperTopOffset: number;
  spotlightHeight: number;
  spotlightLeftOffset: number;
  spotlightTopOffset: number;
  spotlightWidth: number;
}

export const getPositions = (target: HTMLElement, isMobile: boolean, windowWidth: number): Positions => {
  const { top, left, right, height, width } = getBoundingRectOfNudgeTarget(target);
  const padding = remsToPx(spotlightPadding);
  const spotlightTopOffset = top - padding;
  const spotlightLeftOffset = left - padding;
  const spotlightHeight = height + (padding * 2);
  const spotlightWidth = width + (padding * 2);
  const arrowTopOffset = spotlightTopOffset + spotlightHeight + remsToPx(arrowTopMargin);
  const arrowLeft = spotlightLeftOffset + remsToPx(arrowLeftMargin);
  const contentWrapperTopOffset = arrowTopOffset
    + remsToPx(isMobile ? arrowMobileHeight : arrowDesktopHeight)
    + remsToPx(contentMarginTop);
  const contentWrapperRight = windowWidth - right - padding;
  const closeButtonLeft = left + spotlightWidth;
  const closeButtonTopOffset = contentWrapperTopOffset - remsToPx(closeButtonDistanceFromContent);

  return {
    arrowLeft,
    arrowTopOffset,
    closeButtonLeft,
    closeButtonTopOffset,
    contentWrapperRight,
    contentWrapperTopOffset,
    spotlightHeight,
    spotlightLeftOffset,
    spotlightTopOffset,
    spotlightWidth,
  };
};

// Target may have display set to `contents` so we are calculatings rect for children of passed target.
const getBoundingRectOfNudgeTarget = (target: HTMLElement) => {
  // Starting values depends on if we checks for Math.max or Math.min
  let top = 9999;
  let bottom = 0;
  let left = 9999;
  let right = 0;

  for (const child of Array.from(target.children)) {
    const rect = child.getBoundingClientRect();
    top = Math.min(top, rect.top);
    bottom = Math.max(bottom, rect.bottom);
    left = Math.min(left, rect.left);
    right = Math.max(right, rect.right);
  }

  return {
    height: bottom - top,
    left,
    right,
    top,
    width: right - left,
  };
};

export const usePositions = (isMobile: boolean) => {
  const [windowWidth] = useDebouncedWindowSize();
  const [positions, setPositions] = React.useState<Positions | null>(null);
  const studyGuides = useSelector(hasStudyGuides);
  const isEnabled = useSelector(studyGuidesEnabled);

  React.useEffect(() => {
    const document = assertDocument();
    const target = document.querySelector(`#${nudgeStudyToolsTargetId}`) as HTMLElement | null;
    if (isEnabled && target) {
      // Make sure that we calculate positions with body overflow set to hidden
      // because it causes scrollbar to hide which results with different positions.
      const prevOverflow = document.body.style.overflow;
      document.body.style.overflow = 'hidden';
      setPositions(getPositions(target, isMobile, windowWidth));

      // Resets to the value from before calculations. We want this style change to be handled
      // directly in the component.
      document.body.style.overflow = prevOverflow;
    }
    return () => setPositions(null);
  }, [studyGuides, isEnabled, windowWidth, isMobile]);

  return positions;
};

export const getCounterCookie = () => {
  return Number(Cookies.get(cookieNudge.counter) || 0);
};

export const getDateCookie = () => {
  const lastShownDate = Cookies.get(cookieNudge.date);
  return lastShownDate ? new Date(lastShownDate) : undefined;
};

export const passedTimeInterval = () => {
  const lastShownDate = getDateCookie();
  return !lastShownDate
    ? true
    : (Date.now() - lastShownDate.getTime()) > timeIntervalBetweenShowingNudgeInMs
  ;
};

export const getPageCounterCookie = () => {
  return Number(Cookies.get(cookieNudge.pageCounter) || 0);
};

export const incrementPageCounterCookie = () => {
  const counter = getPageCounterCookie();
  Cookies.set(cookieNudge.pageCounter, (counter + 1).toString());
};

export const shouldDisplayNudgeStudyTools = (): boolean => {
  const counter = getCounterCookie();
  const numberOfPagesOpenedByUser = getPageCounterCookie();

  return counter < nudgeStudyToolsShowLimit
    && passedTimeInterval()
    && (numberOfPagesOpenedByUser >= nudgeStudyToolsMinPageLimit);
};

// Set required cookies and reset opened page counter
export const setNudgeStudyToolsCookies = () => {
  const now = new Date();
  const counter = getCounterCookie();
  Cookies.set(cookieNudge.counter, (counter + 1).toString());
  Cookies.set(cookieNudge.date, now.toString());
  Cookies.remove(cookieNudge.pageCounter);
};

export const useIncrementPageCounter = () => {
  const page = useSelector(pageSelector);
  return React.useEffect(() => {
    const counter = getPageCounterCookie();
    if (page && counter <= nudgeStudyToolsMinPageLimit) { incrementPageCounterCookie(); }
  }, [page]);
};
