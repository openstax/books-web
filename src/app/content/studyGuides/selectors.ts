import { GetHighlightsColorsEnum, HighlightColorEnum } from '@openstax/highlighter/dist/api';
import { createSelector } from 'reselect';
import { getSortedSummaryHighlights } from '../highlights/utils';
import * as parentSelectors from '../selectors';
import { filterCounts, getSelectedHighlightsLocationFilters } from '../utils/sharedHighlightsUtils/selectorsUtils';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.studyGuides
);

export const studyGuidesEnabled = createSelector(
  localState,
  (state) => state.isEnabled
);

export const studyGuidesSummary = createSelector(
  localState,
  (state) => state.summary
);

export const hasStudyGuides = createSelector(
  studyGuidesSummary,
  (summary) => summary.highlights !== null
    && summary.totalCountsPerPage
    && Object.keys(summary.totalCountsPerPage).length > 0
);

export const studyGuidesOpen = createSelector(
  localState,
  (state) => state.open
);

export const studyGuidesIsLoading = createSelector(
  localState,
  (state) => state.loading
);

export const studyGuidesPagination = createSelector(
  studyGuidesSummary,
  (summary) => summary.pagination
);

export const hasMoreResults = createSelector(
  studyGuidesPagination,
  (pagination) => Boolean(pagination)
);

export const studyGuidesSummaryHighlights = createSelector(
  localState,
  (state) => state.summary.highlights
);

const summaryFilters = createSelector(
  localState,
  (_state) => null
);

export const totalCountsPerPage = createSelector(
  localState,
  (state) => state.summary.totalCountsPerPage || {}
);

const rawSummaryLocationFilters = createSelector(
  parentSelectors.highlightLocationFilters,
  (locationFilters) =>  Array.from(locationFilters.keys())
);

const rawSummaryColorFilters = createSelector(
  summaryFilters,
  (_filters) => [
    GetHighlightsColorsEnum.Blue,
    GetHighlightsColorsEnum.Green,
    GetHighlightsColorsEnum.Pink,
    GetHighlightsColorsEnum.Purple,
    GetHighlightsColorsEnum.Yellow,
  ] as unknown as HighlightColorEnum[]
);

export const summaryLocationFilters = createSelector(
  rawSummaryLocationFilters,
  (selectedLocations) => new Set(selectedLocations)
);

export const summaryColorFilters = createSelector(
  rawSummaryColorFilters,
  (selectedColors) => new Set(selectedColors)
);

const selectedHighlightLocationFilters = createSelector(
  parentSelectors.highlightLocationFilters,
  summaryLocationFilters,
  getSelectedHighlightsLocationFilters
);

export const filteredCountsPerPage = createSelector(
  totalCountsPerPage,
  selectedHighlightLocationFilters,
  summaryColorFilters,
  filterCounts
);

export const orderedStudyGuidesHighlights = createSelector(
  studyGuidesSummaryHighlights,
  parentSelectors.highlightLocationFilters,
  getSortedSummaryHighlights
);
