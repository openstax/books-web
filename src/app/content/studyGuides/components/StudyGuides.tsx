import { HTMLElement } from '@openstax/types/lib.dom';
import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components/macro';
import { typesetMath } from '../../../../helpers/mathjax';
import Loader from '../../../components/Loader';
import { useServices } from '../../../context/Services';
import theme from '../../../theme';
import { assertWindow } from '../../../utils';
import SectionHighlights, {
  HighlightsChapterWrapper,
  HighlightSection,
  HighlightWrapper
} from '../../components/SectionHighlights';
import allImagesLoaded from '../../components/utils/allImagesLoaded';
import HighlightsWrapper from '../../styles/HighlightsWrapper';
import LoaderWrapper from '../../styles/LoaderWrapper';
import * as selectors from '../selectors';
import StudyGuidesListElement from './StudyGuidesListElement';
import StudyGuidesPrintButton from './StudyGuidesPrintButton';

// tslint:disable-next-line: variable-name
const StudyGuides = ({ className }: { className: string }) => {
  const orderedStudyGuides = useSelector(selectors.orderedSummaryStudyGuides);
  const isLoading = useSelector(selectors.summaryIsLoading);
  const container = React.useRef<HTMLElement>(null);
  const services = useServices();

  React.useLayoutEffect(() => {
    if (container.current) {
      services.promiseCollector.add(allImagesLoaded(container.current));
      services.promiseCollector.add(typesetMath(container.current, assertWindow()));
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps, ignore promiseCollector
  }, [orderedStudyGuides]);

  return <div className={className}>
    <StudyGuidesPrintButton />
    {isLoading ? <LoaderWrapper><Loader large /></LoaderWrapper> : null}
    {orderedStudyGuides && <HighlightsWrapper ref={container}>
      {orderedStudyGuides.map((highlightData) => {
        return <SectionHighlights
          key={highlightData.location.id}
          highlightDataInSection={highlightData}
          highlightRenderer={(highlight) => (
            <StudyGuidesListElement
              key={highlight.id}
              highlight={highlight}
            />
          )}
        />;
      })}
    </HighlightsWrapper>}
  </div>;
};

export default styled(StudyGuides)`
  ${HighlightsChapterWrapper} {
    ${theme.breakpoints.mobile`
      max-width: 90%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `}

    @media print {
      padding-left: 0;
    }
  }

  ${HighlightWrapper} {
    @media print {
      margin-left: 0;
    }
  }

  ${HighlightSection} {
    ${theme.breakpoints.mobile`
      padding-left: 2rem;
    `}

    @media print {
      background: ${theme.color.neutral.darkest}
    }
  }

  display: contents;
`;
