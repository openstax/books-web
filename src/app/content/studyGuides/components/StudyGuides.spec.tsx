import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { typesetMath } from '../../../../helpers/mathjax';
import createTestServices from '../../../../test/createTestServices';
import createTestStore from '../../../../test/createTestStore';
import { book as archiveBook, page, pageInChapter } from '../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../test/mocks/osWebLoader';
import * as Services from '../../../context/Services';
import MessageProvider from '../../../MessageProvider';
import { Store } from '../../../types';
import { assertWindow } from '../../../utils';
import { receiveBook, receivePage } from '../../actions';
import allImagesLoaded from '../../components/utils/allImagesLoaded';
// Temporary import from /highlights directory until we make all this logic reusable and move it to content/
import { highlightLocationFilters } from '../../selectors';
import { formatBookData } from '../../utils';
import { stripIdVersion } from '../../utils/idUtils';
import { getHighlightLocationFilterForPage } from '../../utils/sharedHighlightsUtils';
import { receiveStudyGuidesHighlights } from '../actions';
import { StudyGuidesHighlights } from '../../../types';
import StudyGuides from './StudyGuides';

const hlBlue = { id: 'hl1', color: HighlightColorEnum.Blue, annotation: 'hl1' };
const hlGreen = { id: 'hl2', color: HighlightColorEnum.Green, annotation: 'hl' };
const hlPink = { id: 'hl3', color: HighlightColorEnum.Pink, annotation: 'hl' };
const hlPurple = { id: 'hl4', color: HighlightColorEnum.Purple, annotation: 'hl' };
const hlYellow = { id: 'hl5', color: HighlightColorEnum.Yellow };

describe('StudyGuides', () => {
  const book = formatBookData(archiveBook, mockCmsBook);
  let store: Store;
  let services: ReturnType<typeof createTestServices>;

  beforeEach(() => {
    store = createTestStore();

    store.dispatch(receiveBook(book));
    store.dispatch(receivePage({...page, references: []}));

    services = createTestServices();
  });

  it('properly display summary highlights', () => {
    const state = store.getState();
    const pageId = stripIdVersion(page.id);
    const locationFilters = highlightLocationFilters(state);
    const location = getHighlightLocationFilterForPage(locationFilters, pageInChapter);

    const summaryHighlights = {
      [pageId]: {
        [pageId]: [
          hlBlue,
          hlGreen,
          hlPink,
          hlPurple,
          {...hlYellow, color: null} /* for coverage for css when color was not found */,
        ],
      },
      [location!.id]: {
        [pageInChapter.id]: [hlBlue, hlGreen],
      },
    } as StudyGuidesHighlights;

    store.dispatch(receiveStudyGuidesHighlights(summaryHighlights, null));

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <StudyGuides />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('add promises to promise collector at render', () => {
    const container = assertWindow().document.createElement('div');
    const spyPromiseCollectorAdd = jest.spyOn(services.promiseCollector, 'add');

    const pageId = stripIdVersion(page.id);
    const locationFilters = highlightLocationFilters(store.getState());
    const location = getHighlightLocationFilterForPage(locationFilters, pageInChapter);
    const summaryHighlights = {
      [pageId]: {
        [pageId]: [
          hlBlue,
          hlGreen,
          hlPink,
          hlPurple,
          {...hlYellow, color: null} /* for coverage for css when color was not found */,
        ],
      },
      [location!.id]: {
        [pageInChapter.id]: [hlBlue, hlGreen],
      },
    } as StudyGuidesHighlights;

    store.dispatch(receiveStudyGuidesHighlights(summaryHighlights, null));

    renderer.create(<Provider store={store}>
      <Services.Provider value={services} >
        <MessageProvider>
          <StudyGuides />
        </MessageProvider>
      </Services.Provider>
    </Provider>, { createNodeMock: () => container });

    expect(spyPromiseCollectorAdd).toHaveBeenCalledWith(allImagesLoaded(container));
    expect(spyPromiseCollectorAdd).toHaveBeenCalledWith(typesetMath(container, assertWindow()));
  });
});
