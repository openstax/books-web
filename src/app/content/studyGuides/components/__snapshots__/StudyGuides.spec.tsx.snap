// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudyGuides properly display summary highlights 1`] = `
<div
  className="StudyGuides-sc-1d60zwp-0 cAwzzu"
>
  <div
    className="HighlightsWrapper-vefi0f-0 kuspas"
  >
    <div
      className="SectionHighlights__HighlightsChapterWrapper-c51y19-0 kARwht"
    >
      <div
        className="SectionHighlights__HighlightsChapter-c51y19-1 HowFS"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<span class=\\"os-number\\">10</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 5</span>",
          }
        }
        data-testid="chapter-title"
      />
    </div>
    <div
      className="SectionHighlights__HighlightWrapper-c51y19-2 gFcHNj"
    >
      <div
        className="SectionHighlights__HighlightSection-c51y19-3 cYxWVd"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<span class=\\"os-text\\">Test Page 6</span>",
          }
        }
        data-testid="section-title"
      />
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        >
          hl1
        </div>
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 dSbxG"
          color="blue"
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl1"
          />
        </div>
      </div>
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        >
          hl
        </div>
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 dATmSN"
          color="green"
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl2"
          />
        </div>
      </div>
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        >
          hl
        </div>
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 kiJznw"
          color="pink"
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl3"
          />
        </div>
      </div>
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        >
          hl
        </div>
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 isUPcu"
          color="purple"
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl4"
          />
        </div>
      </div>
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        />
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 iZJlhi"
          color={null}
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl5"
          />
        </div>
      </div>
    </div>
    <div
      className="SectionHighlights__HighlightsChapterWrapper-c51y19-0 kARwht"
    >
      <div
        className="SectionHighlights__HighlightsChapter-c51y19-1 HowFS"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<span class=\\"os-number\\">12</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 6</span>",
          }
        }
        data-testid="chapter-title"
      />
    </div>
    <div
      className="SectionHighlights__HighlightWrapper-c51y19-2 gFcHNj"
    >
      <div
        className="SectionHighlights__HighlightSection-c51y19-3 cYxWVd"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<span class=\\"os-text\\">Test Page 7</span>",
          }
        }
        data-testid="section-title"
      />
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        >
          hl1
        </div>
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 dSbxG"
          color="blue"
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl1"
          />
        </div>
      </div>
      <div
        className="StudyGuidesListElement__HighlightOuterWrapper-sc-5qgyda-0 cEYBxy"
      >
        <div
          className="StudyGuidesListElement__HighlightAnnotation-sc-5qgyda-1 bjHUkM"
        >
          hl
        </div>
        <div
          className="StudyGuidesListElement__HighlightContentWrapper-sc-5qgyda-3 dATmSN"
          color="green"
        >
          <div
            className="summary-highlight-content StudyGuidesListElement__HighlightContent-sc-5qgyda-2 jwHLiF"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            data-highlight-id="hl2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
